decisions/2022-12-12_patient_and_program_transfer.md

```diff
@@ -0,0 +1,92 @@
+# Patient and program transfers
+
+- _Date_: 2022-12-12
+- _Deciders_:
+- _Status_:
+- _Outcome_:
+
+## Context
+
+Patients and programs document need to be accessible at multiple stores.
+This KDD tries to answer the following questions:
+
+- How to search, access and edit patient across multiple stores?
+- How to manage programs at multiple stores and transfer programs between stores?
+- How to manage program permission for different stores?
+
+**Summary of existing mSupply patient editing**:
+
+- Patients are only editable on the home store
+- Only patients from the `name` table can be edited (other names can't be edited). Note, patient push needs to be implemented in omSupply.
+- The patient's home store is identified in the `name` table by the `supplying_store_id` column
+- Its currently not directly supported to change a patient's home store
+
+**Summary of existing mSupply patient search and patient visibility**:
+
+- The endpoint `/patient-search` looks up all patients on central (no access control). The endpoint returns all patient information for a found patient.
+- The client can choose a patient and call `/nsj` to request a new `name_store_join` in order to make the patient visible in a store.
+- The client doesn't wait for sync but directly stores the fetched `name` and `name_store_join` records (client might receive entries again through sync)
+- patient is now visible on the store and syncs to the store (because of the `name_store_join`)
+
+**Transfer Programs Requirements**
+In general, a patient can have a list of programs and each program can be treated at different stores/clinic.
+A patient may change the clinic where the program is treated.
+Thus we need to be able to record the program store affiliation.
+Potentially a program can be treated at multiple stores(?).
+Its probably also useful to have a clinic transfer history.
+
+There are three types of transfers:
+
+- External in transfers: program transfer from an external clinic to mSupply
+- External out transfers: program is not manage by mSupply anymore
+- Internal transfers, a program is transferred to a different store/clinic within mSupply
+
+**Patient visit**
+A patient might have a program encounter at another store.
+To be clarified first:
+
+- user at visited store doesn't has access to the program data?
+- user can add a program encounter but not access other patient program and encounter data?
+
+## Options
+
+### Option 1
+
+**Patient editing:**
+
+- When changing a patient document the patient `name` is updated and pushed to central
```

Comment:
We could also have a standard form for core `name` record (doesn't have to be in JSON schema), and the document are the extensions. It seems easier then mapping document back/forward for name record

Reply:
There is a form that mixes core name record entries and other entries though, so we would have to merge/unmerge the two data sources at a later stage...

Reply:
Not sure why we would need any merging/unmerging, the `core` patient info (the stuff in the name table), can be a standard (not json) form, and the extra data collection specific to a country can be captured with document.

Reply:
There is a patient json UI schema for editing both data at the same time. JSONForms only works on one single JSON blob and for this the data has must be "merged/unmerged". By having all patient data in the document you also get the full information in the patient document history.